R/utility.R

```diff
@@ -65,7 +65,13 @@ readJson <- function(x, simplifyVector = TRUE, ...) {
 }
 
 writeJson <- function(x, file, auto_unbox = TRUE, pretty = TRUE, ...) {
-  jsonlite::write_json(x, path = file, auto_unbox = auto_unbox, pretty = pretty, ...)
+  printingArgs <- list(...)
```

Comment:
This part shouldn't be needed if `na="null"` is propagated via `...` through all the function calls

Reply:
which part? The `list(...)` call? If so, I've tried `na=..1` directly in the `jsonlite::writejson` call and that works, too. The purpose of the list call is to accommodate multiple arguments if they are eventually passed down in a future update. 

Also, I'm using the `...` here so I can be free to modify the upstream `export` calls with additional arguments to `jsonlite::write_json`. In this change, I am only using an addiitonal argument upstream with `exportModels`, hence the ifelse logic. The other `export` functions don't have the `na="null"` argument.

Reply:
The whole thing  seems unnecessary to me. `writeJson()` already passes additional arguments to `jsonlite::write_args()` via `...`. I don't see the advantage of capturing the arguments in a list and then manually adding them. Can you re-try with the original version of `writeJson()` and see if `na = "null"` is passed correctly? You can even do `debug(jsonlite::write_json)` to enter the interactive debugger and print the value of `na`

Reply:
The argument is being passed down from `exportModels()` not from `writeJson()`. The chain is like this:

> 	exportModels
       exportElements (self-referencing function)
						exportElementsWrite (non-nested base case)
							writeJson(…)
								jsonlite::write_json(…)

But yeah, I was hoping I wouldn't have to reference `...` at all and `jsonlite::write_json` would just recognize `na=null`. This was what I checked first without the iflese logic. Didn't work but I'll check again to be sure and get back to you.

Reply:
yeah, it works with the `...` only... Thanks for the review John!